### PR TITLE
Silence `kontrol build` warnings

### PR DESCRIPTION
**Pull request by JuanCoRo**
Depends on https://github.com/runtimeverification/evm-semantics/pull/2497 (which depends on https://github.com/runtimeverification/k/pull/4471)

This PR adds the option `--no-silence-warnings` to `kontrol build`. If `kontrol build` is called without `--no-silence-warnings`, the `non-exhaustive-match` and `missing-syntax-module` warnings will be suppressed.

### CODE REVIEW

File: src/kontrol/cli.py [hunk 1]
```diff
@@ -315,6 +315,13 @@ def parse(s: str) -> list[T]:
         action='store_true',
         help="Do not call 'forge build' during kompilation.",
     )
+    build.add_argument(
+        '--no-silence-warnings',
+        dest='no_silence_warnings',
+        default=None,
+        action='store_true',
+        help='Do not silence K compiler warnings',
```
Comment (palinatolmach) [inline src/kontrol/cli.py:323]:
```suggestion
        help='Do not silence K compiler warnings.',
```
I know we're not very consistent about it, but the rest of `build` help messages seem to end with dots :)

**Review by palinatolmach — APPROVED** [review]
Approved with a very minor comment!

**Comment by palinatolmach** [comment]
Blocked on https://github.com/runtimeverification/kontrol/pull/655 passing the CI.

File: src/kontrol/cli.py (hunk 1 above)
Comment (JuanCoRo) [reply src/kontrol/cli.py:323]: Thanks for the catch!!

**Pull request merged**